web/src/ui.rs

```diff
@@ -33,19 +36,51 @@ impl std::error::Error for FullScreenError {
 }
 
 pub struct WebFileDialogResult {
-    handle: Option<FileHandle>,
+    canceled: bool,
+    file_name: Option<String>,
+    modification_time: Option<DateTime<Utc>>,
     contents: Vec<u8>,
 }
 
 impl WebFileDialogResult {
-    pub async fn new(handle: Option<FileHandle>) -> Self {
+    pub async fn new_pick(handle: Option<FileHandle>) -> Self {
         let contents = if let Some(handle) = handle.as_ref() {
             handle.read().await
         } else {
             Vec::new()
         };
 
-        Self { handle, contents }
+        #[cfg(not(target_arch = "wasm32"))]
+        let file_name = None;
+
+        #[cfg(not(target_arch = "wasm32"))]
+        let modification_time = None;
+
+        #[cfg(target_arch = "wasm32")]
```

Comment:
Please don't interleave those.

Reply:
You mean the targets or the variables? I made the variables not interleaved now, but at the end one of them has to be :/

Reply:
Sorry, I am not sure if you were waiting on a response here. This looks exactly like what I had in mind. :+1: